server/api/scheduler_test.go

```diff
@@ -107,8 +108,152 @@ func (s *testScheduleSuite) TestAPI(c *C) {
+			createdName = ca.name
+		}
+		isPaused, err := handler.GetSchedulePauseState(createdName)
+		c.Assert(err, IsNil)
+		c.Assert(isPaused, Equals, false)
+	}
+
+	// test resume all schedulers.
+	err = postJSON(s.urlPrefix+"/all", pauseArgs)
+	c.Assert(err, IsNil)
+	input["time"] = "0"
+	pauseArgs, err = json.Marshal(input)
+	err = postJSON(s.urlPrefix+"/all", pauseArgs)
+	c.Assert(err, IsNil)
+	for _, ca := range cases {
+		createdName := ca.createdName
+		if createdName == "" {
+			createdName = ca.name
+		}
+		isPaused, err := handler.GetSchedulePauseState(createdName)
+		c.Assert(err, IsNil)
+		c.Assert(isPaused, Equals, false)
+	}
+
+	// delete schedulers.
+	for _, ca := range cases {
+		createdName := ca.createdName
+		if createdName == "" {
+			createdName = ca.name
+		}
+		s.deleteScheduler(createdName, c)
+	}
+
+}
+
+func (s *testScheduleSuite) addScheduler(name, createdName string, body []byte, extraTest func(string, *C), c *C) {
+	if createdName == "" {
+		createdName = name
+	}
+	err := postJSON(s.urlPrefix, body)
+	c.Assert(err, IsNil)
+
+	if extraTest != nil {
+		extraTest(createdName, c)
+	}
+}
+
+func (s *testScheduleSuite) deleteScheduler(createdName string, c *C) {
+	deleteURL := fmt.Sprintf("%s/%s", s.urlPrefix, createdName)
+	err := doDelete(deleteURL)
+	c.Assert(err, IsNil)
+}
+
+func (s *testScheduleSuite) testPauseAndResumeScheduler(name, createdName string, body []byte, extraTest func(string, *C), c *C) {
+	if createdName == "" {
+		createdName = name
+	}
+	err := postJSON(s.urlPrefix, body)
+	c.Assert(err, IsNil)
+	handler := s.svr.GetHandler()
+	sches, err := handler.GetSchedulers()
+	c.Assert(err, IsNil)
+	c.Assert(sches[0], Equals, createdName)
+
+	// test pause.
+	input := make(map[string]interface{})
+	input["time"] = "5"
+	pauseArgs, err := json.Marshal(input)
+	c.Assert(err, IsNil)
+	err = postJSON(s.urlPrefix+"/"+createdName, pauseArgs)
+	c.Assert(err, IsNil)
+	time.Sleep(time.Second)
+	isPaused, err := handler.GetSchedulePauseState(createdName)
+	c.Assert(err, IsNil)
+	c.Assert(isPaused, Equals, true)
+	time.Sleep(time.Second * 4)
+	isPaused, err = handler.GetSchedulePauseState(createdName)
+	c.Assert(isPaused, Equals, false)
+
+	// test resume.
+	input = make(map[string]interface{})
+	input["time"] = "5"
+	pauseArgs, err = json.Marshal(input)
+	c.Assert(err, IsNil)
+	err = postJSON(s.urlPrefix+"/"+createdName, pauseArgs)
+	c.Assert(err, IsNil)
+	input["time"] = "0"
+	pauseArgs, err = json.Marshal(input)
+	err = postJSON(s.urlPrefix+"/"+createdName, pauseArgs)
+	c.Assert(err, IsNil)
+	isPaused, err = handler.GetSchedulePauseState(createdName)
+	c.Assert(err, IsNil)
+	c.Assert(isPaused, Equals, false)
+
+	if extraTest != nil {
+		extraTest(createdName, c)
+	}
+
+	deleteURL := fmt.Sprintf("%s/%s", s.urlPrefix, createdName)
+	err = doDelete(deleteURL)
+	c.Assert(err, IsNil)
 }
 
 func (s *testScheduleSuite) testAddAndRemoveScheduler(name, createdName string, body []byte, extraTest func(string, *C), c *C) {
```

Comment:
I think `testPauseAndResumeScheduler` covers this test?

Reply:
should i remove `testAddAndRemoveScheduler`? this is a previous test func.

Reply:
I think we can.

Reply:
ok, finished.